MapboxNavigation/Error.swift

```diff
@@ -0,0 +1,20 @@
+import Foundation
+
+
+/**
+ Custom Error code key used in returned in `voiceController(_:didInterrupt:with:)`.
+ */
+public let SpokenInstructionErrorCodeKey = "SpokenInstructionErrorCodeKey"
+
+extension NSError {
+    /**
+     Creates a custom `Error` object.
+     */
+    public convenience init(localizedFailureReason: String, detailedFailureReason: String, code: MapboxNavigationError = .defaultError) {
+        self.init(domain: MGLErrorDomain, code: code.rawValue, userInfo: [
+            NSLocalizedFailureReasonErrorKey: localizedFailureReason,
+            SpokenInstructionErrorCodeKey: detailedFailureReason
```

Comment:
`MBSpokenInstructionErrorCodeKey` should be set to an enumeration that a developer can easily `switch` on, not a string.

/ref https://github.com/mapbox/mapbox-navigation-ios/pull/800#discussion_r149804673